### PR TITLE
Add an option to add a list of excluded PRs to the changelog

### PR DESCRIPTION
**Pull request by taketwo**
Also fixes a tag formatting bug introduced in #3821.

### CODE REVIEW

File: .dev/scripts/generate_changelog.py [hunk 1]
```diff
@@ -149,7 +148,7 @@ def make_pr_bullet_point(pr, prefix=None):
     if prefix in ("modules", "categories"):
         tags = "".join(["[" + k + "]" for k in pr[prefix]])
     if tags:
-        tags = "**" + tags + "**"
+        tags = "**" + tags + "** "
 
     return f"* {tags}{pr['title']} [{ref}]"
```
Comment (kunaltyagi) [inline .dev/scripts/generate_changelog.py]:
```suggestion
        tags = "**" + tags + "** "

    return f"* {tags} {pr['title']} [{ref}]"
```

File: .dev/scripts/generate_changelog.py [hunk 2]
```diff
@@ -222,4 +226,9 @@ def generate_module_section(key, prs):
     for k in MODULES.keys():
         clog.extend(generate_module_section(k, selected_prs))
 
+    if args.with_excluded:
+        clog += ["\n### Miscellaneous PRs excluded from changelog\n"]
```
Comment (kunaltyagi) [inline .dev/scripts/generate_changelog.py]: Is it possible to be more uniform in adding '\n'? L221 and L225 don't add one on the end, just the beginning. Adding only on the end always (and adding empty newlines) makes more sense to me while reading.

File: .dev/scripts/generate_changelog.py [hunk 3]
```diff
@@ -184,6 +183,12 @@ def generate_module_section(key, prs):
     cache_grp = parser.add_mutually_exclusive_group()
     cache_grp.add_argument("--save", type=Path, help="Save PR data fetched from GitHub")
     cache_grp.add_argument("--load", type=Path, help="Load PR data from a file")
+    parser.add_argument(
+        "--with-excluded",
```
Comment (kunaltyagi) [inline .dev/scripts/generate_changelog.py]: `--with-misc` or `--keep-unimportant`, etc. seem more self-explanatory

File: .dev/scripts/generate_changelog.py (hunk 1 above)
Comment (taketwo) [reply .dev/scripts/generate_changelog.py]: Why? If `tags` is empty we'll get a double-space.

File: .dev/scripts/generate_changelog.py (hunk 3 above)
Comment (taketwo) [reply .dev/scripts/generate_changelog.py]: This was an allusion to now-removed "changelog: exclude" label. But you are right, `--with-misc` is better.

File: .dev/scripts/generate_changelog.py (hunk 1 above)
Comment (kunaltyagi) [reply .dev/scripts/generate_changelog.py]: I thought this was a formatting change, but it's not strictly a formatting change.

File: .dev/scripts/generate_changelog.py (hunk 2 above)
Comment (taketwo) [reply .dev/scripts/generate_changelog.py]: I see the point that this does not look uniform, but what's the alternative that results in the same output? Can you give me a patch?

Comment (kunaltyagi) [reply .dev/scripts/generate_changelog.py]: Heading needs an extra empty line around it. For that, we can use a simple class
```py
class Heading(list):
    def __init__(self, title_str: str):
            super().__init__(['\n', title_str, '\n'])
```

Comment (taketwo) [reply .dev/scripts/generate_changelog.py]: Yes, but sometimes a heading comes after another heading. In such cases, we don't want a double empty line between them.

To be honest though, I think this is worth neither yours nor my time. This is just a helper script in `.dev/` folder.

**Pull request merged**